js/content/community.js

```diff
@@ -291,6 +292,260 @@ let CommentHandler = (function(){
+                        c += `${tab}${tab}[td][/td]\n`;
+                        w--;
+                    }
+
+                    g_textarea.collectFromEachSelectedLine(function(line) {
+                        n++;
+                        return `${tab}[tr]\n${tab}${tab}[td]${line}[/td]\n${c}${tab}[/tr]`;
+                    }, ``, ``);
+
+                    let r = ``;
+                    while (h > n) {
+                        r += `\n${tab}[tr]\n${tab}${tab}[td][/td]\n${c}${tab}[/tr]`;
+                        h--;
+                    }
+
+                    g_textarea.wrapSelection("", r);
+                    g_textarea.wrapSelection("[table]\n", "\n[/table]");
+                });
+            };
+
+            BBCode_HyperlinkSelection = function() {
+                let text = g_textarea.getSelection();
+                let Modal = ShowConfirmDialog("URL",
+                    `<div class="commentthread_entry_quotebox">
+                        <textarea class="commentthread_textarea" id="es_url" rows="1">
+                    </textarea></div>`
+                );
+                
+                let url = "";
+                $J("#es_url").on("keydown paste input", function(e) {
+                    let code = e.keyCode || e.which;
+                    if (code == 13) {
+                        Modal.Dismiss();
+                        BBCode_MakeURLFromSelection(url, text);
+                        return;
+                    }
+
+                    url = $J("#es_url").val();
+                });
+
+                Modal.done(() => BBCode_MakeURLFromSelection(url, text));
+            };
+
+            $J(textAreaSelector).each(function(i, elem) {
+                let textarea = $J(elem);
+                textarea.addClass("es_textarea");
+
+                let parent = textarea.parents(".commentthread_entry_quotebox, .forumtopic_reply_entry, .commentthread_comment_edit_textarea_ctn");
+                let grandparent = $J(parent).parents(".commentthread_entry, .forum_newtopic_box, .commentthread_comment_editcontrols");
+                let hidden = grandparent.is(":hidden");
+                let leftmargin = 5 + parseFloat($J(parent).css("margin-left"));
+                if (leftmargin === 5) {
+                    let avatar = $J(parent).prev(".playerAvatar");
+                    leftmargin = 5 + parseFloat(avatar.css("margin-left")) + parseFloat(avatar.css("width")) + parseFloat(avatar.css("margin-right"));
+                }
+
+                $J(grandparent).prepend(
+                    `<div style="margin-left: ${leftmargin}px; margin-bottom: 5px;${hidden ? `` : ` display: none;`}" class="es_editor">
+                        <a class="btn_grey_black btn_small_thin es_editor_plus" href="javascript:BBCode_H1Selection();">
+                            <span><img style="vertical-align: middle;" src="https://steamcommunity-a.akamaihd.net/public/images/sharedfiles/guides/format_header1.png"></span>
```

Comment:
I don't think it's *currently* possible to access any steamcommunity features without https. We use "//" for the protocol to avoid the `Mixed Content` warnings if it turns out that it *is* possible to access the page this is injected into on http://

Reply:
Yeah, I know about "//" and I know it's better, but everywhere else `https` is used as well. I'm just doing it for consistency sake.